timelock-server/src/test/java/com/palantir/atlasdb/timelock/clock/ReversalDetectingClockServiceTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.clock;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import org.junit.After;
+import org.junit.Test;
+
+public class ReversalDetectingClockServiceTest {
+
+    private static final String SERVER_NAME = "foo";
+
+    private final ClockService delegate = mock(ClockService.class);
+    private final ClockSkewEvents events = mock(ClockSkewEvents.class);
+    private final ReversalDetectingClockService clock = new ReversalDetectingClockService(delegate, SERVER_NAME,
+            events);
+
+    @After
+    public void after() {
+        verifyNoMoreInteractions(events);
+    }
+
+    @Test
+    public void doesNotLogIfClockDoesNotGoBackwards() {
+        when(delegate.getSystemTimeInNanos())
+                .thenReturn(1L)
+                .thenReturn(2L)
+                .thenReturn(3L);
+
+        clock.getSystemTimeInNanos();
```

Comment:
(Jumping on this as we want to get this change in for 0.53.0. I'm pretty sure the intention is to call it thrice - @nziebart feel free to shout/change it back if that's wrong!)